viewing_party/party.py

```diff
@@ -1,23 +1,248 @@
+        
+        most_watched_count = 0
+        most_watched_genre = ''
+
+        # iterate through genre list, counting how many times each genre appears
+        for genre in watched_genre_list:
+            genre_count = watched_genre_list.count(genre)
+            # update most_watched_genre and genre count with whatever comes back highest
+            if genre_count > most_watched_count:
+                most_watched_count = genre_count
+                most_watched_genre = genre
+
+        return most_watched_genre
+    else:
+        return None
+        
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    """
+    Creates list of movies that user has watched but friends haven't.
+
+    Parameters: user data, including user's watched list, and friends watched lists
+
+    Returns: new list of movie information dictionaries
+    """
+    # create empty list to store unique movies
+    user_unique_watched = []
+
+    # iterate over user's watched list
+    for i in range(len(user_data["watched"])):
+        # set flag to signal if we're still looking
+        searching_for_movie = True
+
+        # set current title we're looking for
+        looking_for_title = user_data["watched"][i]["title"]
+
+        while searching_for_movie:
+            # iterate through friends list
+            for friend in user_data["friends"]:
+                friend_index = user_data["friends"].index(friend)
+                # iterate through every movie on friend's watch list
+                if searching_for_movie:
+                    for num in range(len(user_data["friends"][friend_index]["watched"])):
+                        # if title matches title we're looking for, break out of loop and start looking for next title
+                        # once a title has been found once, we don't need to see if it's been watched by multiple friends
+                        if user_data["friends"][friend_index]["watched"][num]["title"] == looking_for_title:
+                            searching_for_movie = False
+                            break
+                        else:
+                            continue
+                # if no longer searching for movie, break out of loops until searching movie is changed
+                elif not searching_for_movie:
+                    break
+            # if after iterating through all friends watched lists we are still looking, add movie to the unique list
+            if searching_for_movie:
+                user_unique_watched.append(user_data["watched"][i])
+                # then change flag to switch to next movie
+                searching_for_movie = False
+        # continue until all movies on user's watch list have been checked. 
+        continue
+
+    return user_unique_watched       
+
+
+def get_friends_unique_watched(user_data):

```

Comment:
Let's chat through this function too. At the core, what we want to do is:
For each movie in each friend's watched list
    check if the current movie exists in our watched list
        if it does not, add the movie to our unique list

Another way to solve this is a similar approach to the one shown in the feedback for `get_unique_watched` above - we could start by flattening all our friend's watched movies data into a single list we can loop over.